### ISSUE TITLE
inputs.sysstat took longer to collect

### ISSUE THREAD

**Issue by zhou-mfk** [influxdata/telegraf#3840]
Hi,
I encountered the following problem when using telegarf.

![image](https://user-images.githubusercontent.com/15190461/36714108-bfa2414a-1bcb-11e8-81ce-584a25d94e99.png)

sometimes 11 s ,
/usr/lib64/sa/sadc -S DISK 11 2

I think this command caused the timeout.

E! Error in plugin [inputs.sysstat]: took longer to collect than collection interval (10s)

### Relevant telegraf.conf:
[global_tags]
[agent]
  interval = "10s"
  round_interval = true
  metric_batch_size = 1000
  metric_buffer_limit = 10000
  collection_jitter = "3s"
  flush_interval = "10s"
  flush_jitter = "0s"
  precision = ""
  debug = true
  quiet = false
  logfile = ""
  hostname =""
  omit_hostname = false
[[inputs.cpu]]
  percpu = true
  totalcpu = true
  collect_cpu_time = false
[[inputs.disk]]
  ignore_fs = ["tmpfs", "devtmpfs", "devfs"]
[[inputs.diskio]]
[[inputs.kernel]]
[[inputs.mem]]
[[inputs.processes]]
[[inputs.swap]]
[[inputs.system]]
[[inputs.kernel_vmstat]]
[[inputs.linux_sysctl_fs]]
[[inputs.net]]
[[inputs.sysstat]]
  sadc_path = "/usr/lib64/sa/sadc"
  [inputs.sysstat.options]
    -d = "disk"

### System info:

Telegraf v1.3.4 (git: release-1.3 7bbd3daa98d5931c6e9025a98105092b0722de0d)

Red Hat Enterprise Linux Server release 6.5 (Santiago)

### Log info:
/var/log/telegraf/telegraf.log
2018-02-27T06:09:47Z E! Error in plugin [inputs.sysstat]: took longer to collect than collection interval (10s)
2018-02-27T06:10:00Z E! Error in plugin [inputs.sysstat]: took longer to collect than collection interval (10s)
2018-02-27T06:10:12Z E! Error in plugin [inputs.sysstat]: took longer to collect than collection interval (10s)
2018-02-27T06:10:25Z E! Error in plugin [inputs.sysstat]: took longer to collect than collection interval (10s)
2018-02-27T06:10:37Z E! Error in plugin [inputs.sysstat]: took longer to collect than collection interval (10s)
2018-02-27T06:10:49Z E! Error in plugin [inputs.sysstat]: took longer to collect than collection interval (10s)
2018-02-27T06:11:02Z E! Error in plugin [inputs.sysstat]: took longer to collect than collection interval (10s)
2018-02-27T06:11:15Z E! Error in plugin [inputs.sysstat]: took longer to collect than collection interval (10s)
2018-02-27T06:11:29Z E! Error in plugin [inputs.sysstat]: took longer to collect than collection interval (10s)
2018-02-27T06:11:43Z E! Error in plugin [inputs.sysstat]: took longer to collect than collection interval (10s)
2018-02-27T06:11:56Z E! Error in plugin [inputs.sysstat]: took longer to collect than collection interval (10s)
2018-02-27T06:12:08Z E! Error in plugin [inputs.sysstat]: took longer to collect than collection interval (10s)
2018-02-27T06:12:19Z E! Error in plugin [inputs.sysstat]: took longer to collect than collection interval (10s)
2018-02-27T06:12:31Z E! Error in plugin [inputs.sysstat]: took longer to collect than collection interval (10s)

Please help to see if there is any problem?

Thanks

**Comment by danielnelson** [comment]
Thanks for opening this issue, we need to update this plugin to allow for a configurable timeout.  In the meantime you might try increasing the interval for this plugin and see if it helps:
```
[[inputs.sysstat]]
interval = "20s"
sadc_path = "/usr/lib64/sa/sadc"
[inputs.sysstat.options]
-d = "disk"
```

**Comment by zhou-mfk** [comment]
Thanks ! 
Forgive me for my late reply. Your advice I will test and return the test results to you

**Comment by danielnelson** [comment]
A parameter to force the amount of time spent gathering for this plugin in #4068, which may be helpful for your case.  It will be included in 1.7.0, or you can get it early in the nightly builds:
```
[[inputs.sysstat]]
  sadc_interval = "10s"
```

This doesn't fully solve the issue for the reasons mentioned in https://github.com/influxdata/telegraf/pull/4068#issuecomment-384181382

**Comment by powersj** [comment]
I'm closing this issue as both the configurable timeout and the plugin-specific interval options exist.
